src/plugins/expression_reveal_image/server/plugin.ts

```diff
@@ -29,8 +29,11 @@ export class ExpressionRevealImagePlugin
       SetupDeps,
       StartDeps
     > {
-  public setup(core: CoreSetup, { expressions }: SetupDeps): ExpressionRevealImagePluginSetup {
-    expressions.registerFunction(revealImageFunction);
+  public async setup(
```

Comment:
@crob611 I just want to be sure that we really want it all. Especially if some time ago @pgayvallet migrated most plugins to synchronous lifecycle #89562

Reply:
@alexwizp Yes the plugin lifecycle methods should be synchronous and will be forced to be at some point. I believe there will be a warning on startup if you are using async.